src/components/Recipes/Recipes.tsx

```diff
@@ -16,23 +15,21 @@ const Recipes: React.FC = () => {
   const timeout = useRef();
   const [ isLoading, setIsLoading ] = useState(false);
 
-  const debouncer = useCallback<any>(() => {debounce( 1400, setIsLoading, setQueryString, ingredientList, timeout );}, [ingredientList]);
-
-
   useEffect(() => {
-    if (ingredientList.length > 0) {
-      debounce( 1400, setIsLoading, setQueryString, ingredientList, timeout )
+    if (ingredientList.length) {
```

Comment:
A good sign you might wanna use a reducer action vs a setState is when you need to do multiple setStates in one call / code block — this is a use case for that — for future reference leaving this comment here